python/cuml/solvers/cd.pyx

```diff
@@ -194,7 +194,7 @@ class CD(Base):
         self.tol = tol
         self.shuffle = shuffle
         self.intercept_value = 0.0
-        self.coef_ = None
+        self._coef_ = None   # accessed via estimator.coef_
```

Comment:
Nice catch!
Looks like elasticnet has the same issue... and several other models.

@dantegd - should we do a bulk update of all occurrences of similar patterns? or are others ok somehow? see e.g. cuml/naive_bayes/naive_bayes.py:214....

Reply:
I think it’d be better to open an issue to create a pytest that checks attributes of classes, where if an attribute is a cumlarray it must start with an `_`, that way we will prevent issues like this from happening. @hcho3 what do you think?